cmd/storagenode-updater/main.go

```diff
@@ -96,119 +107,111 @@ func cmdRun(cmd *cobra.Command, args []string) (err error) {
-		tempArchive, err := ioutil.TempFile(os.TempDir(), serviceName)
 		if err != nil {
-			return errs.New("cannot create temporary archive: %v", err)
+			return checker.Error.Wrap(err)
 		}
-		defer func() { err = errs.Combine(err, os.Remove(tempArchive.Name())) }()
 
-		log.Println("start downloading", downloadURL, "to", tempArchive.Name())
-		err = downloadArchive(ctx, tempArchive, downloadURL)
-		if err != nil {
-			return err
-		}
-		log.Println("finished downloading", downloadURL, "to", tempArchive.Name())
+		if currentVersion.Compare(suggestedVersion) < 0 {
+			tempArchive, err := ioutil.TempFile(os.TempDir(), runCfg.ServiceName)
+			if err != nil {
+				return errs.New("cannot create temporary archive: %v", err)
+			}
+			defer func() { err = errs.Combine(err, os.Remove(tempArchive.Name())) }()
 
-		extension := filepath.Ext(binaryLocation)
-		if extension != "" {
-			extension = "." + extension
-		}
+			log.Println("start downloading", downloadURL, "to", tempArchive.Name())
+			err = downloadArchive(ctx, tempArchive, downloadURL)
+			if err != nil {
+				return err
+			}
+			log.Println("finished downloading", downloadURL, "to", tempArchive.Name())
 
-		dir := filepath.Dir(binaryLocation)
-		backupExec := filepath.Join(dir, serviceName+".old."+currentVersion.String()+extension)
+			extension := filepath.Ext(runCfg.BinaryLocation)
+			if extension != "" {
+				extension = "." + extension
+			}
 
-		if err = os.Rename(binaryLocation, backupExec); err != nil {
-			return err
-		}
+			dir := filepath.Dir(runCfg.BinaryLocation)
+			backupExec := filepath.Join(dir, runCfg.ServiceName+".old."+currentVersion.String()+extension)
 
-		err = unpackBinary(ctx, tempArchive.Name(), binaryLocation)
-		if err != nil {
-			return err
-		}
+			if err = os.Rename(runCfg.BinaryLocation, backupExec); err != nil {
+				return err
+			}
 
-		downloadedVersion, err := binaryVersion(binaryLocation)
-		if err != nil {
-			return err
-		}
+			err = unpackBinary(ctx, tempArchive.Name(), runCfg.BinaryLocation)
+			if err != nil {
+				return err
+			}
 
-		if suggestedVersion.Compare(downloadedVersion) != 0 {
-			return errs.New("invalid version downloaded: wants %s got %s", suggestedVersion.String(), downloadedVersion.String())
-		}
+			downloadedVersion, err := binaryVersion(runCfg.BinaryLocation)
+			if err != nil {
+				return err
+			}
 
-		log.Println("restarting service", serviceName)
-		err = restartSNService(serviceName)
-		if err != nil {
-			return errs.New("unable to restart service: %v", err)
-		}
-		log.Println("service", serviceName, "restarted successfully")
+			if suggestedVersion.Compare(downloadedVersion) != 0 {
+				return errs.New("invalid version downloaded: wants %s got %s", suggestedVersion.String(), downloadedVersion.String())
+			}
 
-		// TODO remove old binary ??
-	} else {
-		log.Printf("%s version is up to date\n", serviceName)
+			log.Println("restarting service", runCfg.ServiceName)
+			err = restartSNService(runCfg.ServiceName)
```

Comment:
Let's track somewhere that we should make some changes in `restartSNService` to not only return the error returned by `exec.Command.Output`, the output too because it may contain information on what went wrong as we observed when I helped to find the error in the test raised by the CI.

Also, I would change `Output` by `CombinedOutput`.